Services/Mail/classes/class.ilMailTemplatePlaceholderResolver.php

```diff
@@ -24,62 +24,36 @@
  */
 class ilMailTemplatePlaceholderResolver
 {
-    public function __construct(protected ilMailTemplateContext $context, protected string $message)
+    protected Mustache_Engine $mustache_engine;
+
+    public function __construct(Mustache_Engine $mustache_engine)
     {
+        $this->mustache_engine = $mustache_engine;
     }
 
+    /**
+     * @param ilMailTemplateContext $context
+     * @param string $message
+     * @param ilObjUser|null $user
+     * @param array $contextParameters
+     * @param $replaceEmptyPlaceholders boolean
+     * @return string
```

Comment:
Most of the parameter descriptions are obsolete because there is no added value

- Please remove them except for `$contextParameters`
- Tryp to specify the shape of `$contextParameters`